lib/faraday/adapter/em_http.rb

```diff
@@ -70,10 +70,9 @@ def configure_ssl(options, env)
 
         # Reads out timeout settings from env into options
         def configure_timeout(options, env)
-          timeout, open_timeout = request_options(env)
-                                  .values_at(:timeout, :open_timeout)
-          options[:connect_timeout] = options[:inactivity_timeout] = timeout
-          options[:connect_timeout] = open_timeout if open_timeout
+          req = request_options(env)
+          options[:inactivity_timeout] = request_timeout(:read, req)
```

Comment:
This is a good thing, a boundary between EM and Faraday.

And the granularity has increased!